syscall/devices/hostsock/hostsock.c

```diff
@@ -945,6 +945,18 @@ static int _hostsock_ioctl(oe_fd_t* sock_, unsigned long request, uint64_t arg)
     if (!sock)
         OE_RAISE_ERRNO(OE_EINVAL);
 
+    /*
+     * MUSL uses the TIOCGWINSZ ioctl request to determine whether the file
+     * descriptor refers to a terminal device. This request cannot be handled
+     * by Windows hosts, so the error is handled on the enclave side. This is
+     * the correct behavior since sockets are not terminal devices.
+     */
+    switch (request)
+    {
+        default:
+            OE_RAISE_ERRNO(OE_ENOTTY);
```

Comment:
Hello @thomasten and thank you for your contribution. It seems this line will always be reached, which would prevent the call to ``oe_syscall_ioctl_ocall()``. Is this intentional? If so, then I think it would block ``ioctl`` requests on sockets like ``FIONBIO`` and ``FIONREAD``. Did I understand the change correctly?

Reply:
I copied this from hostfs/hostepoll ioctl. I think the intention of this code is that request types should be added when needed. But for fs and epoll there doesn't seem to be a need.

I tested this with OE's tests and with our products and everything worked, so I assumed that there's also no need for any request types for hostsock. However, if you know some request types that you consider important, I'd be happy to add those.

Note that for FIONBIO and FIONREAD requests, `arg` is a pointer. So without my change these requests are still broken.